internal/usagestats/growth.go

```diff
@@ -105,3 +74,89 @@ FROM
+       FROM event_logs
+      WHERE name IN ('InstallBrowserExtensionCTAShown', 'InstallBrowserExtensionCTAClicked' )
+        AND argument->>'page' IN ('file', 'search')
+        AND DATE_TRUNC('month', timestamp) = DATE_TRUNC('month', $1::timestamp)
+ )
+ WITH data_by_month_and_user AS (
+     SELECT name,
+            user_id,
+            DATE_TRUNC('month', timestamp) AS month,
+            argument->>'page' AS page
+       FROM event_logs
+      WHERE name IN ('InstallBrowserExtensionCTAShown', 'InstallBrowserExtensionCTAClicked' )
+        AND argument->>'page' IN ('file', 'search')
+        AND DATE_TRUNC('month', timestamp) = DATE_TRUNC('month', $1::timestamp)
+      GROUP BY user_id
+ )
+ SELECT COUNT(*) FILTER (
+                  WHERE name = 'InstallBrowserExtensionCTAShown'
+                    AND page = 'file') AS user_count_who_saw_bext_cta_on_file_page,
+        COUNT(*) FILTER (
+                  WHERE name = 'InstallBrowserExtensionCTAClicked'
+                    AND page = 'file') AS user_count_who_clicked_bext_cta_on_file_page,
+        COUNT(*) FILTER (
+                  WHERE name = 'InstallBrowserExtensionCTAShown'
+                    AND page = 'search') AS user_count_who_saw_bext_cta_on_search_page,
+        COUNT(*) FILTER (
+                  WHERE name = 'InstallBrowserExtensionCTAClicked'
+                    AND page = 'search') AS user_count_who_clicked_bext_cta_on_search_page
+   FROM data_by_month_and_user,
+        COUNT(*) FILTER (
+                  WHERE name = 'InstallBrowserExtensionCTAShown'
+                    AND page = 'file') AS bext_cta_displays_on_file_page,
+        COUNT(*) FILTER (
+                  WHERE name = 'InstallBrowserExtensionCTAClicked'
+                    AND page = 'file') AS bext_cta_clicks_on_file_page,
+        COUNT(*) FILTER (
+                  WHERE name = 'InstallBrowserExtensionCTAShown'
+                    AND page = 'search') AS bext_cta_displays_on_search_page,
+        COUNT(*) FILTER (
+                  WHERE name = 'InstallBrowserExtensionCTAClicked'
+                    AND page = 'search') AS bext_cta_clicks_on_search_page
+   FROM data_by_month
+`
+	var (
+		userCountWhoSawBextCtaOnFilePage       int32
+		userCountWhoClickedBextCtaOnFilePage   int32
+		userCountWhoSawBextCtaOnSearchPage     int32
+		userCountWhoClickedBextCtaOnSearchPage int32
+		bextCtaDisplaysOnFilePage              int32
+		bextCtaClicksOnFilePage                int32
+		bextCtaDisplaysOnSearchPage            int32
+		bextCtaClicksOnSearchPage              int32
+	)
+	if err := db.QueryRowContext(ctx, query, timeNow()).Scan(
+		&userCountWhoSawBextCtaOnFilePage,
+		&userCountWhoClickedBextCtaOnFilePage,
+		&userCountWhoSawBextCtaOnSearchPage,
+		&userCountWhoClickedBextCtaOnSearchPage,
+		&bextCtaDisplaysOnFilePage,
+		&bextCtaClicksOnFilePage,
+		&bextCtaDisplaysOnSearchPage,
+		&bextCtaClicksOnSearchPage,
+	); err != nil {
+		return nil, err
+	}
+
+	return &types.CTAMetrics{
```

Comment:
@vdavid @erzhtor can we nest these so we can potentially add stats for more CTAs in this JSON later? Something like below:
```suggestion
	return &types.CTAMetrics{
	  BrowserExtensionCTA: {
	    DisplayedOnFilePage:
	      UserCount:
	      EventCount:
	    DisplayedOnSearchPage:
	      UserCount:
	      EventCount:
	    ClickedOnFilePage:
	      UserCount:
	      EventCount:
	    ClickedOnSearchPage:
	      UserCount
	      EventCount:
	```

Reply:
We do something similar for SearchUsage: https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/internal/usagestats/aggregated_search.go?L144-152&utm_source=VSCode-1.4.0#L140:6-140:15